appcelerator/titanium_mobile
# chore(android): rebuilt modules for 9.0.0

**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-25726
- https://jira.appcelerator.org/browse/TIMOB-27724
- https://jira.appcelerator.org/browse/TIMOB-27725
- https://jira.appcelerator.org/browse/TIMOB-27729
- https://jira.appcelerator.org/browse/TIMOB-27730

---
**Summary:**
- [TIMOB-25726](https://jira.appcelerator.org/browse/TIMOB-25726) Have "ti.playservices" module prompt end-user to install/update install Play Services if needed.
  * Will not execute "app.js" until installed/updated by end-user.
  * Feature can be disabled by adding following property to "tiapp.xml" file.
```xml
<ti:app xmlns:ti="http://ti.appcelerator.org">
  <property name="ti.playservices.validate.on.startup" type="bool">false</property>
</ti:app>
```
- [TIMOB-27724](https://jira.appcelerator.org/browse/TIMOB-27724) Rebuild "ti.map" module.
  * Migrated to gradle and AndroidX.
  * Added `x86_64` architecture support.
  * Updates maps library version to `17.0.0`.
- [TIMOB-27725](https://jira.appcelerator.org/browse/TIMOB-27725) Rebuild "ti.facebook" module.
  * Migrated to gradle and AndroidX.
  * Added `x86_64` architecture support.
  * Updated Facebook library version from `5.4.0` to `5.13.0`.
  * Removed deprecated `InviteDialog`. (No longer works on Facebook's end.)
  * Removed `onAppCreate()` method. (Facebook now auto-initializes itself.)
  * Logs deprecation warnings for `LoginButton` properties `publishPermissions` and `readPermissions`. Must use `permissions` property instead.
- [TIMOB-27729](https://jira.appcelerator.org/browse/TIMOB-27729) Rebuild "ti.playservices" module.
  * Migrated to gradle and AndroidX.
  * Added `x86_64` architecture support.
  * Updated Google Play Services core library version from `16.1.5` to `17.1.0`.
- [TIMOB-27730](https://jira.appcelerator.org/browse/TIMOB-27730) Rebuild "ti.webdialog" module.
  * Migrated to gradle and AndroidX.
  * Added `x86_64` architecture support.

---
**Test 1:**
_(You will need to acquire a maps and facebook app id to proceed.)_
1. Download [kitchensink-v2](https://github.com/appcelerator/kitchensink-v2).
2. Build and run on Android.
3. Tap on the "Phone" tab.
4. Tap on the "Maps" row.
5. Verify that a map appears.
6. Press the "Back" button.
7. Tap on the "Mashups" tab.
8. Tap on the "Facebook" row.
9. Tap on the "Login" button.
10. Verify that the Facebook login screen appears.

---
**Test 2:**
1. Create a Classic app project.
2. Add the "ti.webdialog" module as shown below to the "tiapp.xml" file.
3. Copy the below "app.js" code to the project.
4. Build and run on Android.
5. In the app, tap on the "Open Web Dialog" button.
6. Verify that a new window appears displaying the Appcelerator website.

tiapp.xml
```xml
<ti:app xmlns:ti="http://ti.appcelerator.org">
	<modules>
		<module>ti.webdialog</module>
	</modules>
</ti:app>
```

app.js
```javascript
var WebDialog = require('ti.webdialog');
WebDialog.addEventListener('open', function (event) {
	console.log('open: ' + JSON.stringify(event));
});
WebDialog.addEventListener('close', function (event) {
	console.log('close: ' + JSON.stringify(event));
});

var window = Ti.UI.createWindow();
var openButton = Ti.UI.createButton({ title: 'Open Web Dialog' });
openButton.addEventListener('click', function () {
	WebDialog.open({
		url: 'https://appcelerator.com',
		title: 'Hello World',
		tintColor: 'red',
		barColor: 'green',
		barCollapsingEnabled: false,
		dismissButtonStyle: WebDialog.DISMISS_BUTTON_STYLE_CLOSE
	});
});
window.add(openButton);
window.open();
```
